storagemarket/provider_loop.go

```diff
@@ -78,17 +76,15 @@ func (p *Provider) processDealRequest(deal *types.ProviderDealState) (ok bool, r
 	if err != nil {
 		cleanup()
 
-		reason = err.Error()
-		return
+		return false, err.Error(), nil
 	}
 
 	// tag the storage required for the deal in the staging area
 	err = p.storageManager.Tag(p.ctx, deal.DealUuid, deal.Transfer.Size)
 	if err != nil {
 		cleanup()
 
-		reason = err.Error()
-		return
+		return false, err.Error(), nil
```

Comment:
```suggestion
		return false, "server error", err
```

Reply:
@dirkmc leaving this as `return false, err.Error(), nil` as at the moment we don't differentiate between expected conditions and errors within the `StorageManager`.

This is similar to `lack of funds` in the `FundManager`, although while we can assume that `lack of funds` is a condition which should not happen, and we can return a generic `server error` to the client in this case, I don't think we should do that when the miner has its staging area overloaded -- in this case, this is not really a `server error`, but a miner-operating-at-capacity condition which should be communicated back to the client.

Reply:
@dirkmc leaving this as return false, err.Error(), nil as at the moment we don't differentiate between expected conditions and errors within the StorageManager.

This is similar to lack of funds in the FundManager, although while we can assume that lack of funds is a condition which should not happen, and we can return a generic server error to the client in this case, I don't think we should do that when the miner has its staging area overloaded -- in this case, this is not really a server error, but a miner-operating-at-capacity condition which should be communicated back to the client.